nexus/test-utils/src/lib.rs

```diff
@@ -138,247 +140,626 @@ pub async fn test_setup<N: NexusServer>(
     .await
 }
 
-pub async fn test_setup_with_config<N: NexusServer>(
-    test_name: &str,
-    config: &mut omicron_common::nexus_config::Config,
-    sim_mode: sim::SimMode,
-    initial_cert: Option<Certificate>,
-) -> ControlPlaneTestContext<N> {
-    let start_time = chrono::Utc::now();
-    let logctx = LogContext::new(test_name, &config.pkg.log);
-    let log = &logctx.log;
+struct RackInitRequestBuilder {
+    services: Vec<nexus_types::internal_api::params::ServicePutRequest>,
+    datasets: Vec<nexus_types::internal_api::params::DatasetCreateRequest>,
+    internal_dns_config: internal_dns::DnsConfigBuilder,
+}
```

Comment:
This is basically a builder to support accurate information to DNS and to the RSS handoff request.